laws/src/main/scala/cats/laws/discipline/arbitrary.scala

```diff
@@ -130,7 +130,20 @@ object arbitrary extends ArbitraryInstances0 with ScalaVersionSpecific.Arbitrary
       a <- A.arbitrary
     } yield NonEmptyMap((k, a), fa))
 
-  implicit def cogenNonEmptyMap[K: Order: Cogen, A: Order: Cogen]: Cogen[NonEmptyMap[K, A]] =
+  @deprecated("Preserved for bincompat", "2.9.0")
+  def cogenNonEmptyMap[K, A](kOrder: Order[K],
+                             kCogen: Cogen[K],
+                             aOrder: Order[A],
+                             aCogen: Cogen[A]
+  ): Cogen[NonEmptyMap[K, A]] = {
+    implicit val orderingK: Order[K] = kOrder
+    implicit val cogenK: Cogen[K] = kCogen
+    implicit val cogenA: Cogen[A] = aCogen
+
+    Cogen[SortedMap[K, A]].contramap(_.toSortedMap)
+  }
```

Comment:
Can we delegate to the new method?
```suggestion
  ): Cogen[NonEmptyMap[K, A]] =
  cogenNonEmptyMap(kOrder, kCogen, aCogen)
```